File: libccif/src/bp/cdlGroupId.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:17 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "cdlGroupId.h"

#ifdef org_himalay_ccif_cdlGroupId__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
cdlGroupId::cdlGroupId() {
    references= 0;


    // groupId
    groupId= (ui16)0;
}
// Destructor
cdlGroupId::~cdlGroupId() {
    // groupId

}

int cdlGroupId::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // groupId
    {
        groupId=istream.readUI16();
        retVal+=2;
    }
    return retVal;
}

int cdlGroupId::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read groupId
    {
        groupId=istream.readUI16();
        retVal+=2;
    }

    return retVal;
}

int cdlGroupId::write(DataOutputStream& ostream) { // throws IOException


    int retVal= 0;
    // write groupId
    ostream.writeUI16(groupId);
    retVal +=2;

    return retVal;
}

int cdlGroupId::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("cdlGroupId\n");
    dc.increaseIndent();
    int retVal= 0;
    // write groupId
    dc.indent();
    dc.getPs().print("groupId=");
    dc.getPs().println((long)groupId);
    dc.decreaseIndent();
    return retVal;
}


// Getter for groupId
//ui16 cdlGroupId::getGroupId()
//{
//return this->groupId;
//}

// Setter for groupId
//void cdlGroupId::setGroupId(ui16 val)
//{
//this->groupId= val;
//}

int cdlGroupId::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_ccif_cdlGroupId__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::cdlGroupId* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::cdlGroupId* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code

File: ccif/src/java/other/org/himalay/ccif/cdlGroupId.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:33:06 EST 2016")

public  class cdlGroupId extends BinStruct implements PublicBinMsg {




    // members variables
    // groupId
    public int groupId ;

    public cdlGroupId () // throws Exception
    {
        init();
    }

    private void init()
    {
        // Initialize groupId

    }

    public int readNoHeader(DataInputStream istream) throws IOException
    {

        return read(istream);
    }

    public int read(DataInputStream istream) throws IOException
    {
        preRead();
        int retVal= 0;

        // read groupId
        {
            groupId=istream.readUnsignedShort();
            retVal+=2;
        }

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException
    {
        preWrite();
        int retVal= 0;



        // write groupId
        ostream.writeShort(groupId);
        retVal +=2;
        postWrite();
        return retVal;
    }

    public int dump(DumpContext dc) throws IOException
    {
        dc.indent();
        dc.getPs().print("cdlGroupId\n");
        dc.increaseIndent();
        int retVal= 0;
        // write groupId
        dc.indent();
        dc.getPs().println("groupId="+groupId+"(0x"+ Integer.toHexString(groupId) + ")" );
        dc.decreaseIndent();
        return retVal;
    }


    // Getter for groupId
    //public int getGroupId()
    //{
    //    return groupId ;
    //}


    // Setter for groupId
    //public void setGroupId(int val)
    //{
    //    this.groupId= val;
    //}


    public int getSize() throws IOException
    {
        DataOutputStream dos= new DataOutputStream(new NullStream());
        return this.write(dos);
    }



}

// End of code